### PR TITLE
Stm32F7: add external ram config

### PR DESCRIPTION
**Pull request by AlexanderVasiljev**

### CODE REVIEW

File: arch/arm/src/stm32f7/Kconfig [hunk 1]
```diff
@@ -2136,6 +2136,14 @@ endchoice #"ULPI Selection"
...
 endmenu # OTG_HS Config
 
+config STM32F7_EXTERNAL_RAM
+    bool "External RAM on FMC"
+    default n
+    depends on STM32F7_FMC
+    select ARCH_HAVE_HEAP2
+    ---help---
+        In addition to internal SDRAM, external RAM may be available through the FMC.
+
```
Comment (patacongo) [inline arch/arm/src/stm32f7/Kconfig]: Where is this new configuration option used?  It is not used in arch/arm/src/stm32f7 so it does not belong there.

Do you use it in your board?  If so then it belongs in your boards/arm/stm32/<board>/Kconfig

Comment (AlexanderVasiljev) [reply arch/arm/src/stm32f7/Kconfig]: The sole purpose of STM32F7_EXTERNAL_RAM is to select ARCH_HAVE_HEAP2.
ARCH_HAVE_HEAP2 is used in arch/arm/src/stm32f7/stm32_allocateheap.c
There is no other way to select ARCH_HAVE_HEAP2.

It is the same as in stm32h7. Stm32H7 has STM32H7_EXTERNAL_RAM, which just selects ARCH_HAVE_HEAP2.
And Stm32 also has this option. STM32_EXTERNAL_RAM
Only Stm32F7 is missing it.

Comment (tritao) [reply arch/arm/src/stm32f7/Kconfig]: In our own board, which also uses F7, we also had to hack around `ARCH_HAVE_HEAP2` to get external RAM to work.
So I think there's some value to this change, if this were merged, we would switch to this option.

**Comment by AlexanderVasiljev** [comment]
> Configuration is improperly scoped, improperly name, and in the wrong Kconfig file.
> 
> Move unused configuration option to the /board/arm/stm32ft/your-board/Kconfig
> 
> Rename the option to CONFIG_YOURBOARD_EXTERNAL_RAM

I would like to insist to apply this patch as is.
1) STM32F7_EXTERNAL_RAM will select ARCH_HAVE_HEAP2, which is used in arch/arm/src/stm32f7/stm32_allocateheap.c. And there is no other way to select ARCH_HAVE_HEAP2. Now it is not obvious that stm32f7 has already support for second heap in arch/arm/src/stm32f7/stm32_allocateheap.c
2) Stm32 and stm32H7 aleady have the same options: STM32H7_EXTERNAL_RAM and STM32_EXTERNAL_RAM. For the sake of uniformity Stm32F7 should have the same option. Otherwise it is frustrating when migrating from stm32f4 to stm32f7 and not finding such a config.

**Comment by patacongo** [comment]
Okay.  I think it is generally bad usage of Kconfig variables.  But there seems to be a precedence.

File: arch/arm/src/stm32f7/Kconfig (hunk 1 above)
Comment (patacongo) [inline arch/arm/src/stm32f7/Kconfig]: Please replace spaces with TABs to match usage in other Kconfig files.

**Comment by patacongo** [comment]
Thanks for the update.  I am supposed to wait for all of the PR-triggered checks to complete, then I will (squash) merge your changes.

**Pull request merged**